src/components/hooks/use-client-information.js

```diff
@@ -1,26 +1,28 @@
-import { useState, useEffect } from 'react'
+import { useState, useEffect, useRef } from 'react'
 import { getClientInformation, getDomain } from 'common/utility'
 
 export const useClientInformation = () => {
     const [client_information, setClientInformation] = useState(false)
-    const [current_client_information, setCurrentClientInformation] = useState(false)
+    const callback_ref = useRef()
+
+    const setCurrentClientInformation = () => {
+        const current_client_information = getClientInformation(getDomain())
+        if (JSON.stringify(current_client_information) != JSON.stringify(client_information)) {
+            setClientInformation(current_client_information)
+        }
+    }
+
+    useEffect(() => {
+        callback_ref.current = setCurrentClientInformation
+    })
```

Comment:
This will rerun the operations inside `useEffect` on every re-render.  Is this intended?

Reply:
Yes, this is to save the latest callback so we can use the updated state in the interval.

Reply:
It is quite a bad practice to go without a dependency array. Maybe as a suggestion, we could use a helper function like this instead

```
const useCallbackRef = (callback) => {
    const callbackRef = useRef(callback)

    useLayoutEffect(() => {
        callbackRef.current = callback
    }, [callback])

    return callbackRef
}
```